fluent/toInterfaceValue_test.go

```diff
@@ -0,0 +1,51 @@
+package fluent_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime/fluent"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+)
+
+var roundTripTestCases = []struct {
+	desc  string
+	value interface{}
+}{
+	{desc: "Number", value: 100},
+	{desc: "String", value: "hi"},
+	{desc: "Bool", value: "hi"},
+	{desc: "Bytes", value: []byte("hi")},
+	{desc: "Map", value: map[string]interface{}{"a": "1", "b": int64(2), "c": 3.14, "d": true}},
+	{desc: "Array", value: []string{"a", "b", "c"}},
+}
+
+func TestRoundTrip(t *testing.T) {
+	for _, testCase := range roundTripTestCases {
+		t.Run(testCase.desc, func(t *testing.T) {
+			c := qt.New(t)
+			n, err := fluent.Reflect(basicnode.Prototype.Any, testCase.value)
+			c.Assert(err, qt.IsNil)
+			out, err := fluent.ToInterface(n)
+			c.Assert(err, qt.IsNil)
+			outJson, err := json.Marshal(out)
+			qt.Check(t, err, qt.IsNil)
+			qt.Check(t, outJson, qt.JSONEquals, testCase.value)
```

Comment:
nit: you should be able to use `qt.DeepEquals`, I think. the only other change you'd need is for the Array test case to use `[]interface{}` as its type.

Reply:
👍 I was wondering if that would work after I made this change. `qt` is pretty neat.